test/Microsoft.Health.Dicom.Tests.Integration/Persistence/CustomTagStoreTests.STOW.cs

```diff
@@ -0,0 +1,203 @@
+            DicomDataset dataset = Samples.CreateRandomInstanceDataset(studyInstanceUid, seriesInstanceUid, sopInstanceUid);
+            dataset.Add(element);
+            CustomTagLevel level = CustomTagLevel.Study;
+            var customTagEntry = element.Tag.BuildCustomTagEntry(level: level);
+
+            IndexTag indexTag = await AddCustomTag(customTagEntry);
+            try
+            {
+                long watermark = await _indexDataStore.CreateInstanceIndexAsync(dataset, new IndexTag[] { indexTag });
+                Instance instance = await _testHelper.GetInstanceAsync(studyInstanceUid, seriesInstanceUid, sopInstanceUid, watermark);
+                IReadOnlyList<CustomTagDataRow> rows = await _testHelper.GetCustomTagDataAsync(dataType, indexTag.CustomTagStoreEntry.Key, instance.StudyKey);
+                Assert.Single(rows);
+                Assert.Equal(watermark, rows[0].Watermark);
+                Assert.Equal(expectedValue, rows[0].TagValue);
+            }
+            finally
+            {
+                await _customTagStore.DeleteCustomTagAsync(customTagEntry.Path, customTagEntry.VR);
+            }
+        }
+
+        public static IEnumerable<object[]> SupportedTypes()
+        {
+            yield return new object[] { CustomTagDataType.StringData, new DicomCodeString(DicomTag.ConversionType, "STRING"), "STRING" };
+            yield return new object[] { CustomTagDataType.DateTimeData, new DicomDateTime(DicomTag.AcquisitionDateTime, DateTime.Parse("2020/2/21")), DateTime.Parse("2020/2/21") };
+            yield return new object[] { CustomTagDataType.DoubleData, new DicomFloatingPointDouble(DicomTag.DopplerCorrectionAngle, 1.0), 1.0 };
+            yield return new object[] { CustomTagDataType.LongData, new DicomSignedLong(DicomTag.ReferencePixelX0, 1), 1L };
+            yield return new object[] { CustomTagDataType.PersonNameData, new DicomPersonName(DicomTag.DistributionNameRETIRED, "abc^abc"), "abc^abc" };
+        }
+
+        private async Task ValidateAdd(CustomTagLevel level)
+        {
+            string studyInstanceUid = TestUidGenerator.Generate();
+            string seriesInstanceUid = TestUidGenerator.Generate();
+            string sopInstanceUid = TestUidGenerator.Generate();
+            DicomDataset dataset = Samples.CreateRandomInstanceDataset(studyInstanceUid, seriesInstanceUid, sopInstanceUid);
+            DicomTag tag = DicomTag.ConversionType;
+            string value = "SYN";
+            dataset.Add(tag, value);
+
+            IndexTag indexTag = await AddCustomTag(tag.BuildCustomTagEntry(level: level));
+            try
+            {
+                long watermark = await _indexDataStore.CreateInstanceIndexAsync(dataset, new IndexTag[] { indexTag });
+                Instance instance = await _testHelper.GetInstanceAsync(studyInstanceUid, seriesInstanceUid, sopInstanceUid, watermark);
+                long? seriesKey = level != CustomTagLevel.Study ? instance.SeriesKey : null;
```

Comment:
Are you validating the sql code that is adding SeriesKey and InstanceKey to the custom table based on tagLevel?

Reply:
these integration tests verify :
* adding new custom tag index tag  
* updating existing data ( on series/study custom tag)